File: TransactionHistory.py
import uuid

from Transaction import Transaction

class TransactionHistory:
    def __init__(self):
        self.id = str(uuid.uuid4())
        self.transactions = []

    def add_transaction(self, amount, transaction_type, destination):
        transaction = Transaction(amount, transaction_type, destination)
        self.transactions.append(transaction)

File: TransactionHistory.js
const { Transaction } = require('./Transaction');
const { uuid } = require('uuidv4');

class TransactionHistory {
    constructor() {
        this.id = uuid();
        this.transactions = [];
    }

    addTransaction(amount, transactionType, destination) {
        const transaction = new Transaction(amount, transactionType, destination);
        this.transactions.push(transaction);
    }
}

module.exports = { TransactionHistory };
